### PR TITLE
Invalid structure for keystore methods

### PR DESCRIPTION
**Pull request by r8d8**
Moved methods for keystore IO into serialize module
Added tests

### CODE REVIEW

File: src/keystore/mod.rs [hunk 1]
```diff
@@ -108,6 +103,19 @@ impl KeyFile {
 
         self.keccak256_mac = keccak256(&v);
     }
+
+    /// Creates seed vectors for `kdf` and `cipher`
+    pub fn crypto_seed(&mut self) {
```
Comment (dulanov) [inline src/keystore/mod.rs]: Better to combine with `new`?

File: src/keystore/mod.rs [hunk 2]
```diff
@@ -108,6 +103,19 @@ impl KeyFile {
...
+        let mut salt: [u8; KDF_SALT_BYTES] = [0; 32];
+        let mut iv: [u8; CIPHER_IV_BYTES] = [0; 16];
+
+        let mut rng = OsRng::new().ok().unwrap();
```
Comment (dulanov) [inline src/keystore/mod.rs]: Change to `.expect`

File: src/keystore/serialize/mod.rs [hunk 3]
```diff
@@ -9,9 +9,12 @@ mod error;
 pub use self::address::try_extract_address;
 use self::crypto::Crypto;
 use self::error::Error;
-use super::Address;
-use super::KeyFile;
-use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
+use super::{Address, KeyFile, PrivateKey};
```
Comment (dulanov) [inline src/keystore/serialize/mod.rs]: `super::core::...`

File: src/keystore/serialize/mod.rs [hunk 4]
```diff
@@ -9,9 +9,12 @@ mod error;
...
+use chrono::prelude::*;
```
Comment (dulanov) [inline src/keystore/serialize/mod.rs]: Avoid `*` in production code, only when more when i don't know 8 or 10 items

File: src/keystore/serialize/mod.rs [hunk 5]
```diff
@@ -68,9 +71,107 @@ impl From<SerializableKeyFile> for KeyFile {
...
+
+/// Creates a new `KeyFile` with a specified `Address`
+///
+/// # Arguments
+///
+/// * `pk` - private core for inserting in a `KeyFile`
+/// * `passphrase` - password for encryption of private core
+/// * `addr` - optional address to be included in `KeyFile`
+///
+pub fn create_keyfile(pk: PrivateKey, passphrase: &str, addr: Option<Address>) -> KeyFile {
```
Comment (dulanov) [inline src/keystore/serialize/mod.rs]: Where is here IO stuff?

File: src/keystore/serialize/mod.rs [hunk 6]
```diff
@@ -68,9 +71,107 @@ impl From<SerializableKeyFile> for KeyFile {
...
+    let mut file = File::create(&path).expect("Expect to create file for KeyFile");
+    let data = json::encode(&kf).expect("Expect to encode KeyFile");
+    file.write_all(data.as_ref()).ok();
+    Ok(file)
+}
+
+/// Creates filename for keystore file in format:
+/// `UTC--yyy-mm-ddThh-mm-ssZ--uuid`
+pub fn get_filename() -> String {
+    format!("UTC--{}--{}", &get_timestamp(), &Uuid::new_v4())
```
Comment (dulanov) [inline src/keystore/serialize/mod.rs]: `Z` should stay here, not in `get_timestamp`, or?

Comment (r8d8) [inline src/keystore/serialize/mod.rs]: Z comes with timestamp, right
Just put full format in doc for clarity.

**Pull request merged**